### PR TITLE
MAILET-103 Refactor RemoveAllMailAttributes

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java [hunk 1]
```diff
@@ -38,25 +39,12 @@
  */
 public class RemoveAllMailAttributes extends GenericMailet {
     
-    /**
-     * Return a string describing this mailet.
-     *
-     * @return a string describing this mailet
-     */
     public String getMailetInfo() {
```
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java]: @Override missing

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java [hunk 2]
```diff
@@ -20,49 +20,43 @@
...
-        Assert.assertTrue(mockedMail.getAttributeNames().hasNext());
+    public void serviceShouldRemoveAllMailAttributes() throws MessagingException {
+        mockedMail = MailUtil.createMockMail2Recipients(null);
+        mockedMail.setAttribute("org.apache.james.test.junit", "true");
 
         mailet.service(mockedMail);
 
-        // check if all was removed
-        Assert.assertFalse(mockedMail.getAttributeNames().hasNext());
+        assertThat(mockedMail.getAttributeNames().hasNext()).isFalse();
```
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: isEmpty

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java [hunk 3]
```diff
@@ -20,49 +20,43 @@
...
     }
 
-    // test if ToProcessor works
     @Test
-    public void testRemoveAllMailAttributes() throws MessagingException {
-        setupMockedMail(null);
-        // check if the mail has a attribute
-        Assert.assertTrue(mockedMail.getAttributeNames().hasNext());
+    public void serviceShouldRemoveAllMailAttributes() throws MessagingException {
+        mockedMail = MailUtil.createMockMail2Recipients(null);
```
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: s/mockedMail/mail/

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java [hunk 4]
```diff
@@ -38,25 +39,12 @@
...
     }
 
-    /**
-     * Removes all attributes on the Mail
-     *
-     * @param mail the mail to process
-     *
-     * @throws MessagingException in all cases
-     */
     public void service(Mail mail) throws MessagingException {
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java]: @override too

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java [hunk 5]
```diff
@@ -20,49 +20,43 @@
...
+        assertThat(mailet.getMailetInfo()).isEqualTo("Remove All Mail Attributes Mailet");
     }
 
-    // test if ToProcessor works
     @Test
-    public void testRemoveAllMailAttributes() throws MessagingException {
-        setupMockedMail(null);
-        // check if the mail has a attribute
-        Assert.assertTrue(mockedMail.getAttributeNames().hasNext());
+    public void serviceShouldRemoveAllMailAttributes() throws MessagingException {
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: maybe you can add a testcase when you have no attribute

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java (hunk 1 above)
Comment (aduprat) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java]: OK

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java (hunk 4 above)
Comment (aduprat) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/RemoveAllMailAttributes.java]: OK

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java (hunk 2 above)
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: OK

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java (hunk 3 above)
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: OK

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java (hunk 5 above)
Comment (aduprat) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveAllMailAttributesTest.java]: OK

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

**Comment by mbaechler** [comment]
👍

**Comment by rouazana** [comment]
:+1:

**Pull request merged**